app/utils/trayWithText.js

```diff
@@ -0,0 +1,262 @@
+const mergeImg = require('merge-img-vwv')
+const pathToImages = 'app/images/app-icons/numbers/generated-numbers/'
+const pathToCircleImages = 'app/images/app-icons/round-clock/'
+const log = require('electron-log')
+const baseImages = 'app/images/app-icons/numbers/'
+
+class TrayWithText {
+  iconWithNumber (
+    imagePath,
+    minutesToLongBreak,
+    darkModeString,
+    monochrome,
+    platform,
+    invertedMonochromeString
+  ) {
+    let minutesOnTray = minutesToLongBreak
+    let minutesAsNumber = 99
+    try {
+      minutesAsNumber = Number(minutesOnTray)
+    } catch (er) {
+      minutesAsNumber = 99
+    }
+    if (minutesAsNumber >= 99) {
+      minutesOnTray = '99'
+    }
+    if (monochrome) {
+      if (platform === 'darwin') {
+        return `${imagePath}traytMacMonochrome${minutesOnTray}Template.png`
+      } else {
+        return `${imagePath}traytMonochrome${invertedMonochromeString}${minutesOnTray}.png`
+      }
+    } else {
+      if (platform === 'darwin') {
+        return `${imagePath}traytMac${darkModeString}${minutesOnTray}.png`
+      } else {
+        return `${imagePath}trayt${darkModeString}${minutesOnTray}.png`
+      }
+    }
+  }
+
+  pathWithCircularIcon (
+    imagePath,
+    minutesToLongBreak,
+    totalLongBreak,
+    darkModeString,
+    monochrome,
+    platform,
+    invertedMonochromeString
+  ) {
+    const minutesOnTray = minutesToLongBreak
+    const timeLeftRatio = minutesOnTray / totalLongBreak
+
+    let pictureNo = '0'
+    if (timeLeftRatio >= 0.875) {
+      pictureNo = '0'
+    } else if (timeLeftRatio < 0.875 && timeLeftRatio > 0.75) {
+      pictureNo = '7'
+    } else if (timeLeftRatio <= 0.75 && timeLeftRatio > 0.625) {
+      pictureNo = '15'
+    } else if (timeLeftRatio <= 0.625 && timeLeftRatio > 0.5) {
+      pictureNo = '22'
+    } else if (timeLeftRatio <= 0.5 && timeLeftRatio > 0.375) {
+      pictureNo = '30'
+    } else if (timeLeftRatio <= 0.375 && timeLeftRatio > 0.25) {
+      pictureNo = '37'
+    } else if (timeLeftRatio <= 0.25 && timeLeftRatio > 0.125) {
+      pictureNo = '45'
+    } else if (timeLeftRatio <= 0.125 && timeLeftRatio > 0.065) {
+      pictureNo = '52'
+    } else {
+      pictureNo = '60'
+    }
+    if (monochrome) {
+      if (platform === 'darwin') {
+        return `${imagePath}traytMacMonochrome${pictureNo}Template.png`
+      } else {
+        return `${imagePath}traytMonochrome${invertedMonochromeString}${pictureNo}.png`
+      }
+    } else {
+      if (platform === 'darwin') {
+        return `${imagePath}traytMac${darkModeString}${pictureNo}.png`
+      } else {
+        return `${imagePath}trayt${darkModeString}${pictureNo}.png`
+      }
+    }
+  }
+
+  async generateNumbers (color = 'b', mac = 'm') {
```

Comment:
There is no test coverage for the icon generation.

Reply:
And image generations are not really used in app. Once generated it's no longer needed. They could only be needed when icon changes for some reason. So I guess it's not needed.

Reply:
Ok, I guess that's fine. I will probably try getting designer  to make something in the future